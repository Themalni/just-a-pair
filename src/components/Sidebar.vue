<template>
  <div class="sidebar-container">
    <h2 class="sidebar-header font-weight-bolder text-large pl-3 pt-4 pb-3">Admin dashboard</h2>
    <div class="sidebar-user mt-3 ml-3 mb-2">Hello, {{ user }}</div>
    <ul class="nav flex-column">
      <li class="nav-item">
        <button class="button-reset text-white ml-3" @click="logout()">
          <md-icon class="icon-sm">logout</md-icon>Logout
        </button>
      </li>
    </ul>
  </div>
</template>

<script>
import { fb } from "@/firebase";

export default {
  name: "Sidebar",
  props: {
    user: {
      type: String
    }
  },
  methods: {
    async logout() {
      await this.$auth.logout();
      this.$router.push("/");
    }
  }
};
</script>
<style lang="scss" scoped>
.sidebar-container {
  width: 240px;
  height: 100vh;
  background-color: $dark-blue;
}
.sidebar-header {
  color: #fff;
  border-bottom: 1px solid rgba(255, 255, 255, 0.5);
}
.btn {
  &.btn-link {
    color: #fff;
  }
}
.sidebar-user {
  color: #fff;
}
</style>
