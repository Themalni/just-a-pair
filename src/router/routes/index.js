export default [{
    path: '/',
    name: 'Home',
    component: () => import( /* webpackChunkName: "Home" */ '@/views/Home')
},
{
    path: '/product/:id',
    name: 'Product',
    component: () => import( /* webpackChunkName: "Product" */ '@/views/Product')
},
{
    path: '/user-dashboard',
    name: 'UserDashboard',
    component: () => import( /* webpackChunkName: "User Dashboard" */ '@/views/user/UserDashboard'),
    meta: {
        authRequired: true
    }
}, {
    path: '/admin-dashboard',
    name: 'AdminDashboard',
    component: () => import( /* webpackChunkName: "Admin Dashboard" */ '@/views/admin/AdminDashboard'),
    meta: {
        authRequired: true
    }
}
]