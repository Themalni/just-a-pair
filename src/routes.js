import ProductView from './components/ProductView.vue'
import HomePage from './components/HomePage.vue'


export default [
    {path: '/', component: HomePage},
    {path: '/product-view/:id', component: ProductView }
]