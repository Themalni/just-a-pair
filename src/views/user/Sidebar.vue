<template>
  <div class="sidebar-container">
    <h3 class="sidebar-header pl-3 pt-4 pb-3">Your profile</h3>
    <div class="separator"></div>
    <div class="sidebar-user mt-3 ml-3 mb-2">Hello, {{ userName }}</div>
    <div class="sidebar-user mt-3 ml-3 mb-2">{{ userEmail }}</div>
    <div class="separator mt-3 mb-4"></div>
    <button class="button-reset text-white ml-3" @click="logout">
      <md-icon class="icon-sm">logout</md-icon>Logout
    </button>
  </div>
</template>

<script>
import { fb } from "@/firebase";
import { mapState } from "vuex";
import { CLEAR_USER } from "@/store/types";
import store from '@/store';

export default {
  name: "Sidebar",

  computed: {
    ...mapState(["user"]),

    userName() {
      return this.user ? this.user.displayName : '';
    },
    userEmail() {
      return this.user ? this.user.email : '';
    }
  },
  methods: {
    logout() {
      fb.auth().signOut()
        .then(() => {
          store.dispatch(CLEAR_USER);

          this.$notify({
            text: 'You logout successfully',
            type: 'success',
            duration: 3000
          });
          this.$router.push("/");
        })
        .catch(() => {
          this.$notify({
            text: 'Logout error',
            type: 'error',
            duration: 3000
          })
        });
    }
  }
};
</script>
<style lang="scss" scoped>
.sidebar-container {
  margin-top: 60px;
  width: 240px;
  height: 93.9vh;
  background-color: $dark-blue;
}
.sidebar-header {
  color: #fff;
}
.btn {
  &.btn-link {
    color: #fff;
  }
}
.sidebar-user {
  color: #fff;
}
.separator {
  border-bottom: 1px solid rgba(255, 255, 255, 0.5);
}
</style>
