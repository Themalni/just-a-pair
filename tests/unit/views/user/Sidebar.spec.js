import { mount, createLocalVue } from '@vue/test-utils';
import Sidebar from "@/views/user/Sidebar";
import mockState from '../../utils/mockState';
import Vuex from 'vuex';
import VueRouter from "vue-router";
import routes from "@/router/routes";

const localVue = createLocalVue();
localVue.use(Vuex);
localVue.use(VueRouter);

describe('Sidebar', () => {
    let wrapper
    let store
    let state
    let router

    beforeEach(() => {
        router = new VueRouter({ routes })
        state = mockState.state
        store = new Vuex.Store({
            state: state
        })

        wrapper = mount(Sidebar, {
            stubs: ['md-icon'],
            store,
            localVue,
            router
        });
    });

    afterEach(() => {
        wrapper.destroy();
    });

    it('renders correctrly', () => {
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('logs out user', async () => {
        const logoutMock = jest.spyOn(wrapper.vm, 'logout');

        await wrapper.find('.logout-btn').trigger('click');

        expect(logoutMock).toHaveBeenCalled();
    })
})