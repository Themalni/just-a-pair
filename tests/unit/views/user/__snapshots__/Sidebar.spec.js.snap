// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar renders correctrly 1`] = `
<div class="sidebar-container">
  <h3 class="sidebar-header pl-3 pt-4 pb-3">Your profile</h3>
  <div class="separator"></div>
  <div class="sidebar-user mt-3 ml-3 mb-2">Hello, John Smith</div>
  <div class="sidebar-user mt-3 ml-3 mb-2">john_doe@gmail.com</div>
  <div class="separator mt-3 mb-4"></div> <button class="logout-btn button-reset text-white ml-3">
    <md-icon-stub class="icon-sm">logout</md-icon-stub>Logout
  </button>
</div>
`;
